README.md

```diff
@@ -1 +1,239 @@
-# node-modules-patterns
\ No newline at end of file
+# Module Patterns in Node
+
+## Overview
+
+This lesson will cover the most common patterns used to create and export a Node module.
+
+## Objectives
+
+1. Define different patterns for Node modules
+1. Describe module.exports= {}
+1. Describe module.exports=function(){}
+1. Describe exports.method = function(){}
+1. Describe exports.object = {}
+
+## Exporting a Module
+
+Module allow developers to build better systems because the code can be separated into different files which are groups into packages or modules. You can write your own module or use a module from an open-source community (some one else created it and made the code available to others free of charge). 
+
+Imagine a scenario where you have an application with 2,000 lines of code. All code is in one file. That's a nightmare because to navigate around in such a file and also different developers will have to be very careful when they collaborate not to introduce conflicts in their changes! What if you split the logic into 20 files which each file having 100 lines of code roughly. Much better right? So how do you write a module in Node.
+
+You can one of the following patterns:
+
+1. `module.exports= {...}`
```

Comment:
just listing the patterns doesn't really mean much if a student doesn't know what their purposes are. it's better to introduce each on individually by defining a situation in which each pattern is useful, then list them all at the end